pkg/operator/ceph/predicate.go

```diff
@@ -77,17 +99,28 @@ func predicateController(ctx context.Context, client client.Client) predicate.Fu
 						// We still want to reconcile the operator manager if the configmap is updated
 						return true
 					}
-				} else if s, ok := e.ObjectNew.(*v1.Secret); ok {
-					if s.Name == admissionControllerAppName {
-						logger.Debug("webhook secret updated, reloading the manager")
+				}
+			} else if s, ok := e.ObjectNew.(*v1.Secret); ok {
+				if s.Name == admissionControllerAppName {
+					logger.Debug("webhook secret updated, reloading the manager")
+					controller.ReloadManager()
+
+					// No need to ask for reconciliation since the context is going to be terminated when
+					// the signal is caught and the reconcile will run when the controller starts.
+					// If the admission controller secret is created or deleted we still need to reload and
+					// the webhook might be enabled or disabled
+					//
+					// The same goes the update, the secret changes we still need to reload the webhook
+					return false
+				}
+			} else if old, ok := e.ObjectOld.(*api.Certificate); ok {
+				if new, ok := e.ObjectNew.(*api.Certificate); ok {
+					if old.Name == certificateName && new.Name == certificateName {
+						logger.Debugf("webhook certificate %s updated reloading the manager to enable the webhook server", certificateName)
```

Comment:
Whenever we reload the manager, let's log info so we know what triggered it, same change suggested for the other reloads as well.
```suggestion
						logger.Infof("webhook certificate %s updated reloading the manager to enable the webhook server", certificateName)
```

Reply:
done